x-pack/plugins/encrypted_saved_objects/index.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Root } from 'joi';
+import { Legacy, Server } from 'kibana';
+
+// @ts-ignore
+import { AuditLogger } from '../../server/lib/audit_logger';
+
+import { CONFIG_PREFIX, PLUGIN_ID, Plugin } from './server/plugin';
+
+export const encryptedSavedObjects = (kibana: any) =>
+  new kibana.Plugin({
+    id: PLUGIN_ID,
+    configPrefix: CONFIG_PREFIX,
+    require: ['kibana', 'elasticsearch', 'xpack_main'],
+
+    config(Joi: Root) {
+      return Joi.object({
+        enabled: Joi.boolean().default(true),
+        encryptionKey: Joi.string().min(32),
+        audit: Joi.object({ enabled: Joi.boolean().default(false) }),
+      }).default();
+    },
+
+    async init(server: Legacy.Server) {
+      const loggerFacade = {
+        fatal: (errorOrMessage: string | Error) => server.log(['fatal', PLUGIN_ID], errorOrMessage),
+        trace: (message: string) => server.log(['debug', PLUGIN_ID], message),
+        error: (message: string) => server.log(['error', PLUGIN_ID], message),
+        warn: (message: string) => server.log(['warning', PLUGIN_ID], message),
+        debug: (message: string) => server.log(['debug', PLUGIN_ID], message),
+        info: (message: string) => server.log(['info', PLUGIN_ID], message),
+      } as Server.Logger;
+
+      const config = server.config();
+      const encryptedSavedObjectsSetup = new Plugin(loggerFacade).setup(
+        {
+          config: {
+            auditLogEnabled: config.get<boolean>(`${CONFIG_PREFIX}.audit.enabled`),
+            encryptionKey: config.get<string>(`${CONFIG_PREFIX}.encryptionKey`),
+          },
+          savedObjects: server.savedObjects,
+          elasticsearch: server.plugins.elasticsearch,
+        },
+        { audit: new AuditLogger(server, PLUGIN_ID) }
```

Comment:
__note:__ it's still not clear how audit logging will look like exactly in new platform - whether it is part of the xpack main plugin or part of the core, so I moved it to `dependencies/plugins` section for now.

Reply:
I think we're likely going to want to use `xpack.security.audit.enabled` to determine whether to perform audit logging because of the other pending changes to the way that audit logging behaves...

Reply:
Got it, thanks! Will make necessary changes.